scripts/charmm_to_fah.py

```diff
@@ -0,0 +1,309 @@
+import copy
+import numpy
+import shutil
+import tempfile
+import numpy as np
+
+import pdbfixer
+from simtk import openmm, unit
+from simtk.openmm import app
+import mdtraj as md
+
+import sys
+
+# ParmEd Imports
+#from parmed.charmm import CharmmPsfFile, CharmmCrdFile, CharmmParameterSet
+from simtk.openmm.app import CharmmPsfFile, CharmmCrdFile, CharmmParameterSet
+
+from parmed import unit as u ### ???
+
+#
+
+
+print("Input PDB structure: 1OL5")
+pdbid = '1OL5'
+runs = 0
+
+output_path = '1OL5-phos-tpx2'
+
+# Source PDB
+pdbfilename = 'step2_solvator.pdb'
+
+print("Source PDB filename: %s" % pdbfilename)
+print("Output directory for mutations: %s" % output_path)
+
+#
+# PARAMETERS
+#
+
+universal_parameter_files = [
+    'toppar/par_all36_prot.prm',
+    'toppar/par_all36_na.prm',
+    'toppar/top_all36_prot.rtf',
+    'toppar/toppar_all36_na_nad_ppi.str',
+    'toppar/toppar_water_ions.str',
+    'step2_solvator.str',
+    'step2.2_ions.prm',
+    'step2.1_waterbox.prm',
+    'step1_pdbreader.str',
+    'step1_labelrot.str',
+    'toppar/toppar_all36_label_spin.str',
+    'step1_pdbreader.str',
+    'toppar/top_all36_cgenff.rtf',
+    'toppar/top_all36_na.rtf',
+    'toppar/toppar_all36_na_rna_modified.str',
+    'toppar/toppar_all36_prot_na_combined.str',
+    'toppar/par_all36_cgenff.prm',
+    'toppar/par_all36_na.prm',
+]
+
+psf_file = 'step2_solvator.psf'
+
+padding = 11.0 * unit.angstroms
+nonbonded_cutoff = 9.0 * unit.angstroms
+nonbonded_method = app.PME
+max_minimization_iterations = 10000
+temperature = 300.0 * unit.kelvin
+pressure = 1.0 * unit.atmospheres
+collision_rate = 5.0 / unit.picoseconds
+barostat_frequency = 50
+timestep = 2.0 * unit.femtoseconds
+nsteps = 50000 # number of steps to take for testing
+ionicStrength = 20 * unit.millimolar
+
+# Verbosity level
+verbose = True
+
+def write_file(filename, contents):
+    with open(filename, 'w') as outfile:
+        outfile.write(contents)
+
+exception_filename = os.path.join(output_path, 'exceptions.out') # to store exceptions
+run_index_filename = os.path.join(output_path, 'run-index.txt') # to store index of which mutants are which
+
+# Create output directory.
+if not os.path.exists(output_path):
+    os.makedirs(output_path)
+
+# Create temporary directory.
+tmp_path = tempfile.mkdtemp()
+print("Working in temporary directory: %s" % tmp_path)
+tmp_path = output_path
+
+# Open file to write all exceptions that occur during execution.
+exception_outfile = open(exception_filename, 'a')
+run_index_outfile = open(run_index_filename, 'a')
+
+
+name = str(runs)
+simulation = None
+
+
+# Load the CHARMM files
+print('Loading CHARMM files...')
+param_files = universal_parameter_files #+ charmm_parameter_files
+params = CharmmParameterSet(*param_files)
+psf = CharmmPsfFile(psf_file)
+pdb = app.PDBFile(pdbfilename)
+
+coords = pdb.positions
+min_crds = [coords[0][0], coords[0][1], coords[0][2]]
+max_crds = [coords[0][0], coords[0][1], coords[0][2]]
+
+for coord in coords:
```

Comment:
This is usually a really poor way to select the box size, since it could literally put atoms on top of each other. Is there no better way to get the box size out of the CHARMM files?

If not, let's enlarge the box by 0.5 A or so in every dimension just in case.